streams/src/main/java/org/apache/kafka/streams/processor/DefaultPartitionGrouper.java

```
@@ -81,18 +80,9 @@ protected int maxNumPartitions(Cluster metadata, Set<String> topics) {
         for (String topic : topics) {
             List<PartitionInfo> partitions = metadata.partitionsForTopic(topic);
 
-            if (partitions.isEmpty()) {
```

For trouble shooting only: maybe we can still check that `!partitions.isEmpty()`, and if yes log an FATAL and throw runtime exception? If we had a bug that still causes `partitions.size() == 0`, then the line 83 below would silently skip assigning the maxNumPartitions update, which would be very hard to capture during debugging.